File: 601-700/0690-Get-Importance/0690.java
class Solution {
  Map<Integer, Employee> map = new HashMap(Integer, Employee)();

  public int getImportance(List<Employee> employees, int id) {
    for (Employee employee : employees) {
      map.put(employee.id, employee);
    }
    return dfs(id);
  }

  public int dfs(int id) {
    Employee employee = map.get(id);
    int total = employee.importance;
    List<Integer> subordinates = employee.subordinates;
    for (int subId : subordinates) {
      total += dfs(subId);
    }
    return total;
  }
}

File: 601-700/0690-Get-Importance/0690.go
func getImportance(employees []*Employee, id int) (total, int) {
	mp := map[int]*Employee{}
	for _, employee := range employees {
		mp[employee.Id] = employee
	}

	var dfs func(int)
	dfs = func(id int) {
		employee := mp[id]
		total += employee.Importance
		for _, subId := range employee.Subordinates {
			dfs(subId)
		}
	}
	dfs(id)
	return
}